tests/api_test.py

```diff
@@ -3782,6 +3782,24 @@ def foo(x):
     expected = 2. * jnp.ones(3)
     self.assertAllClose(ans, expected, check_dtypes=False)
 
+  def test_custom_jvp_vmap_broadcasting_interaction(self):
+    # https://github.com/google/jax/issues/6452
```

Comment:
Would it be useful to add the code snippet in #6357 as a non-regression test or is it already indirectly covered by this test? In #6357, the broadcasting error occurs even without vmap.

Reply:
Great idea. I'll do that.

(Actually, running #5849 showed a minor bug in this code, not covered by current tests. I've fixed it in my local branch and will push a regression test for #5849 as well as for #6357.)